staging/src/k8s.io/component-base/config/types.go

```diff
@@ -78,3 +78,9 @@ type DebuggingConfiguration struct {
 	// enableProfiling is true.
 	EnableContentionProfiling bool
 }
+
+// FeatureGatesConfiguration holds configuration for features.
+type FeatureGatesConfiguration struct {
+	// FeatureGates is a map of feature names to bools that enable or disable alpha/experimental features.
+	FeatureGates map[string]bool
```

Comment:
I'm not sure what the additional level of nesting is getting us. Is this better than a map[string]bool feature gates field directly in a config type?

Reply:
What I'm hoping to get with this is a standard type and validation.

Would it make sense to define each of the features as a field?

If not, maybe we can just do `type FeatureGates map[string]bool` or somehow enforce that `FeatureGatesConfiguration` is always inlined.

Reply:
IMO,  the additional level of nesting is to meet other requirements that may exist, such as disabling all alpha features or enabling all beta features? WDYT?

Reply:
> What I'm hoping to get with this is a standard type and validation.
> Would it make sense to define each of the features as a field?

I don't think we want to _force_ centralization of feature definitions by embedding them in a common type - that's a bigger discussion at least. 

As far as a standard type, we already have `map[string]bool` as a standard (as @liggitt noted), and common helper machinery is already implemented in `k8s.io/kubernetes/vendor/k8s.io/component-base/featuregate/feature_gate.go`.

> IMO, the additional level of nesting is to meet other requirements that may exist, such as disabling all alpha features or enabling all beta features? WDYT?

Today those capabilities are already implemented in the existing mechanism. IMO it doesn't seem necessary to change this.

Reply:
Thanks for the clarifications.

2 nits:

1. Can we add `type FeatureGatesConfig map[string]bool`?
1. Does it make sense to validate that the provided feature names exist at API validation time?

Reply:
It might be useful to add a feature gates helper that can validate a config against the registered features without setting the gates.

I'm not sure how much the typedef gets us.

Reply:
/close

We could add the validation in a new PR.